### PR TITLE
[dom-mc] HPX 1.3.0 with and without profiling options

### PR DESCRIPTION
**Pull request by msimberg**
Adds:
- HPX 1.3.0 with latest Boost, hwloc
- Latest OTF2
- HPX 1.3.0 with latest Boost, hwloc, and profiling options enabled (APEX, PAPI, OTF2, RelWithDebInfo)

Apologies if I've messed up the tags. This is intended for daint-mc.

### CODE REVIEW

**Comment by teojgo** [comment]
Ok to test

**Comment by msimberg** [comment]
```
ERROR: Missing dependencies: papi/5.7.0.1-CrayGNU-19.06, OTF2/.2.2-CrayGNU-19.06 (no easyconfig file or existing module found)
```

Do I need to make a separate pull request for OTF2 first to have that available for this build? And what would be the correct way to specify papi as a dependency? `papi 5.7.0.1` is available as a module, but presumably not with the CrayGNU environment?

**Comment by teojgo** [comment]
@msimberg you can put OTF2 in this PR and use `EXTERNAL_MODULE` for papi.

**Comment by teojgo** [comment]
@msimberg you should also add the papi version here: `production/easybuild/cray_external_modules_metadata.cfg`

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.06-APEX.eb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+toolchainopts = {'pic': True}
+
+builddependencies = [('CMake', '3.14.5', '', True)]
+
+dependencies = [
+    ('Boost', '1.70.0', '-python3'),
+    ('hwloc', '2.0.3', '', True),
+    ('jemalloc', '5.1.0'),
+    ('papi/5.7.0.1', EXTERNAL_MODULE),
+    ('OTF2', '2.2'),
```
Comment (teojgo) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.06-APEX.eb]: Change according to the `hwloc` above since `OTF2` is generic

**Comment by msimberg** [comment]
Thanks @teojgo, updated.

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.06-APEX.eb (hunk 1 above)
Comment (msimberg) [reply easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.06-APEX.eb]: Updated. Thanks!

**Comment by victorusu** [comment]
retest this please

**Comment by msimberg** [comment]
This should be ready to go now. I had to patch HPX to get APEX working as we had a bug which prevents it from checking out the correct version of APEX.

I can remove the changes to `easybuild/cray_external_modules_metadata.cfg` as I ended up not needing them and are unrelated to the HPX additions.

**Pull request merged**